code/obj/flock/structure/annihilator.dm

```diff
@@ -0,0 +1,133 @@
+/obj/flock_structure/annihilator
+	icon_state = "annihilator-off" // placeholder sprites
+	name = "gnesis fountain"
+	desc = "Some sort of gnesis fountain. The gnesis appears very active."
+	flock_desc = "A defense turret that fires high speed gnesis bolts at nearby projectiles, annihilating them."
+	flock_id = "Annihilator"
+	health = 50
+	health_max = 50
+	repair_per_resource = 2.5
+	resourcecost = 100
+	passthrough = TRUE
+
+	compute = 0
+	var/online_compute_cost = 30
+
+	var/powered = FALSE
+	var/checkers_powered = FALSE
+
+	var/checker_radius = 2
+	var/list/projectile_checkers = null
+
+	New(atom/location, datum/flock/F = null)
+		..()
+		var/turf/src_turf = get_turf(src)
+		var/list/turfs = block(locate(max(src_turf.x - src.checker_radius, 1), max(src_turf.y - src.checker_radius, 1), src_turf.z), locate(min(src_turf.x + src.checker_radius, world.maxx), min(src_turf.y + src.checker_radius, world.maxy), src_turf.z))
+
+		src.projectile_checkers = list()
+		for (var/turf/T as anything in turfs)
+			src.projectile_checkers += new /obj/annihilator_projectile_checker(T, src)
+
+		src.info_tag.set_info_tag("Not generating bolt")
+
+	building_specific_info()
+		return !src.powered ? "Not generating bolt." : GET_COOLDOWN(src, "bolt_gen_time") ? "Generation time left: [round(GET_COOLDOWN(src, "bolt_gen_time") / 10)] seconds." : "Bolt ready."
+
+	process(mult)
+		if (!src.flock)
+			if (src.powered)
+				src.update_flock_compute("remove")
+				if (src.checkers_powered)
+					src.power_projectile_checkers("off")
+			src.compute = 0
+			src.powered = FALSE
+		else if (src.flock.can_afford_compute(src.online_compute_cost))
+			src.compute = -src.online_compute_cost
+			if (!src.powered)
+				ON_COOLDOWN(src, "bolt_gen_time", 10 SECONDS)
+				src.update_flock_compute("apply")
+				src.powered = TRUE
+		else if (src.flock.used_compute > src.flock.total_compute() || !src.powered)
+			if (src.powered)
+				src.update_flock_compute("remove")
+				if (src.checkers_powered)
+					src.power_projectile_checkers("off")
+			src.compute = 0
+			src.powered = FALSE
```

Comment:
the first and last if blocks appear to be exactly the same so should be combined

Reply:
They're different, the first case is if there's no Flock, ex. building created without a Flock or it exists without one, last case is for if there's not enough compute to run or the initial case of it being unpowered and not enough compute to run, which needs to be placed after the compute check otherwise the building will never turn on